src/main/scala/codacy/plugins/test/multiple/CheckstyleFormatParser.scala

```diff
@@ -2,31 +2,48 @@ package codacy.plugins.test.multiple
 
 import com.codacy.analysis.core.model.{Parameter, Pattern}
 import com.codacy.plugins.api.results.Result.Level
-import com.codacy.plugins.results.PluginResult
 
 import play.api.libs.json.Json
 
 import com.fasterxml.jackson.core.JsonParseException
 import scala.xml.Elem
+import com.codacy.analysis.core.model.{FileError, Issue, ToolResult}
+import java.nio.file.Paths
+import com.codacy.analysis.core.model.LineLocation
+import com.codacy.plugins.api
 
 private[multiple] object CheckstyleFormatParser {
 
-  def parseResultsXml(root: Elem) = {
+  def parseResultsXml(root: Elem): Seq[ToolResult] = {
     for {
-      fileTag <- root \\ "checkstyle" \\ "file"
-      fileName = fileTag \@ "name"
-      errorsTag <- fileTag \\ "error"
-      line = errorsTag \@ "line"
-      patternId = errorsTag \@ "source"
+      fileTag <- root \\ "checkstyle" \ "file"
+      filePath = Paths.get(fileTag \@ "name")
+      errorsTag <- fileTag \ "error"
       message = errorsTag \@ "message"
-      severity = errorsTag \@ "severity"
-      level = severity match {
-        case "info" => Level.Info
-        case "warning" => Level.Warn
-        case "error" => Level.Err
-        case _ => throw new Exception(s"""$severity is not a valid level. Use one of ["info", "warning", "error"]""")
+      lineAttr = errorsTag \ "@line"
+      patternIdAttr = errorsTag \ "@source"
+      severityAttr = errorsTag \ "@severity"
+      result = if (patternIdAttr.isEmpty && severityAttr.isEmpty) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://app.codacy.com/gh/codacy/codacy-plugins-test/pullRequest?prid=5525832)